docs/en/testing.md

```diff
@@ -0,0 +1,54 @@
+# Unit Testing
+
+Sometimes you may need to test components that need the vue-router to be started. This requires a slightly different test setup.
+
+``` js
+// component.spec.js
+import Vue from 'vue'
+import VueRouter from 'vue-router'
+import Routes from './routes'
+import MyComponent from './my-component'
+
+describe('MyComponent component', () => {
+  it('should pass this dummy test', () => {
+    /* Plug the router into Vue */
+    Vue.use(VueRouter)
+
+    /* This is not a new, but rather a Vue.extend */
+    const App = Vue.extend({
+      /* We assign a ref to our component in order to access it later */
+      template: '<div><my-component :foo="foo" v-ref:mycomponent></my-component></div>',
+      components: { MyComponent },
+      data () {
+        return {
+          foo: 'bar'
+        }
+      },
+      replace: false
+    })
+
+    /* The router is initialized with the abstract option */
+    router = new VueRouter({ abstract: true })
+    router.map(Routes)
+
+    router.start(App, 'body')
```

Comment:
we probably want to create a temporary div and append it to the body instead of mounting directly to the body, especially if we have other unit tests going on on the same page.